speechbrain/lobes/models/huggingface_transformers/llama.py

```diff
@@ -0,0 +1,128 @@
+
+    Example
+    -------
+    >>> model_hub = "meta-llama/Llama-2-7b-chat-hf"
+    >>> save_path = "savedir"
+    >>> model = LLaMA(model_hub, save_path)
+    >>> tokens = torch.tensor([[1, 1]])
+    >>> attention_mask = torch.tensor([[1, 1]])
+    >>> outputs = model(tokens, attention_mask)
+    """
+
+    def __init__(
+        self,
+        source: str,
+        save_path: str,
+        bnb_config: BitsAndBytesConfig = None,
+        freeze: bool = False,
+        pad_token: str = "[PAD]",
+        padding_side: Literal["left", "right"] = "right",
+        weight_precision_load: torch.dtype = torch.bfloat16,
+    ) -> None:
+        self.pad_token = pad_token
+        self.padding_side = padding_side
+        self.source = source
+        self.save_path = save_path
+        self.bnb_config = bnb_config
+
+        if self.bnb_config is not None:
+            logger.info(
+                "LlaMA will be quantised following the given configuration."
+            )
+
+        super().__init__(
+            source=source,
+            save_path=save_path,
+            freeze=freeze,
+            with_casual_lm=True,
+            quantization_config=self.bnb_config,
+            torch_dtype=weight_precision_load,
+        )
+
+        self.load_tokenizer(source=source, pad_token=self.pad_token)
+        self.model.resize_token_embeddings(
+            len(self.tokenizer), pad_to_multiple_of=8
+        )  # TODO: optimal value of pad_to_multiple_of varies? see https://github.com/huggingface/tokenizers/issues/991.
+
+    def forward(self, **kwargs):
+        """This function wraps the HuggingFace forward function. See the HuggingFace documentation of your Llama model of interest to know which
+        parameters to pass, typically the input tokens or embeddings and attention masks.
+
+        Arguments
+        ---------
+        **kwargs : list
+            Please refer to HuggingFace documentation and map it to your Llama model of interest.
+
+        Returns
+        -------
+        output : torch.Tensor
+            This depends on the Llama model. Please refer to the HuggingFace documentation.
+        """
+
+        return self.model(**kwargs)
+
+    def generate(self, **kwargs):
+        """This function wraps the HuggingFace generate function. See the HuggingFace documentation of your Llama model of interest to know which
+        parameters to pass, typically the input tokens or embeddings, attention masks and a transformers.GenerationConfig.
+
+        Arguments
+        ---------
+        **kwargs : list
+            Please refer to HuggingFace documentation and map it to your Llama model of interest.
+
+        Returns
+        -------
+        hyp : torch.Tensor
+            Contains tokenized (indices) outputs.
+        """
+
+        with torch.no_grad():
+            return self.model.generate(**kwargs)
```

Comment:
I am surprised that the default behavior of our `huggingface.py` class is to not call `self.model` methods

Reply:
You just don't like our HF interface :p